### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by wolterbwambale**
In this project, I have implemented the following:-
### Project requirements
- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Abdurahim, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there!
🏁 🚦 ⛪ 🚜 🏃 🌴 🏫 🌳 🏘️

## Highlights 👍  
- Good job so far with creating associations among objects!
- ReadMe looks great!

## Required Changes ♻️
- [ ] In the Student class, we want to establish that _a student belongs to a classroom_. We can achieve that by doing something similar to:

```ruby
class Student < Person
  attr_reader :classroom
  .
  .

  def classroom(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
end
```

Let's read further on the same [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)

- [ ] The Rental class is missing. Kindly define a Rental class whish should be similar to:
```
class Rental
  .
  def initialize(date, book, person)
    .
    .

    @book = book
    book.rentals << self

    @person = person
    # *The rental is adding itself to the @person the 'belongs-to' association
    person.rentals << self
  end
end
```

- [ ] In the _Person_ class, let's ensure a method for adding _rentals_ is present, so as to establish a a _rental_ `belong-to` _person_ relationship, like the code example below:

```ruby
class Person
  attr_accessor :name, :age, :rentals
  .
  .

  def add_rental(book, date)
    Rental.new(date, book, self)
  end

  .
  .
end
```


Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
+  attr_accessor :age, :name, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
-    super()
+    super(name)
     @age = age
     @name = name
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (addod19) [inline person.rb:13]: - [ ] As mentioned by the previous reviewer, we were asked to implement the `add_rental` method, let us kindly implement it

File: student.rb [hunk 2]
```diff
@@ -1,9 +1,10 @@
 require_relative 'person'
 
 class Student < Person
-  def initialize(id, age, classroom, parent_permission: true, name: 'Unknown')
-    super(id, age, parent_permission: parent_permission, name: name)
+  def initialize(classroom, age, name: 'Unknown', parent_permission: true)
+    super(age, name: name, parent_permission: parent_permission)
     @classroom = classroom
+    classroom.add_student(self)
   end
```
Comment (addod19) [inline student.rb:9]: - [ ] As mentioned by the previous code reviewer, we were asked to implement `student belongs to a classroom`, here is how we can achieve the functionality
```
def classroom=(classroom)
   @classroom = classroom
   classroom.students.push(self) unless classroom.students.include?(self)
end

```

**Review by addod19 — CHANGES_REQUESTED** [review]
## Hi @wolterbwambale, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [ ] Nice job adding the rental class as suggested earlier :rocket: 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ibizugbe — APPROVED** [review]
Hi @wolterbwambale 👋🏼👋🏼👋🏼, 

approved

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights✅✅✅✅

- [x] A method to add rental has been implemented in the person class as requested by the previous reviewer👍🏼.
- [x] The `belongs-to` association has been implemented in the student class as requested by the previous reviewer👍🏼.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**